mempool/src/shared_mempool.rs

```diff
@@ -616,12 +622,17 @@ where
     }
 }
 
-fn process_config_update(config_update: OnChainConfigPayload) {
-    // check for VM update
-    if let Ok(_vm_publishing_option) = config_update.get::<VMPublishingOption>() {
-        // TODO restart VM validator
-    }
-    // potentially check for other configs here
+async fn process_config_update<V>(validator: Arc<RwLock<V>>)
+where
+    V: TransactionValidation,
+{
+    // restart VM validator
```

Comment:
I liked previous version better. It's true, that right now we have only one config subscription, but some on-chain config changes might be totally irrelevant to vm_validator and some of them don't require restart. Restarting it unconditionally doesn't seem right

Reply:
ok will revert 🤣

Reply:
My thought on this was that restarting by default is safer. If you add a new irrelevant on-chain config and forget to add it to the "restart blacklist", the consequences are minimal (presumably a small performance hit). But if you add a new relevant config and forget to add it a "restart whitelist", that is potentially much nastier/harder to debug (e.g., might reject transactions that should be accepted after the config change). 

Thoughts? Am I underestimating the perf hit in the first case?